narwhals/_pandas_like/series.py

```diff
@@ -620,6 +620,10 @@ def quantile(
     def zip_with(self: Self, mask: Any, other: Any) -> PandasLikeSeries:
         ser = self._native_series
         mask = validate_column_comparand(ser.index, mask)
+        if isinstance(mask, str) or not isinstance(
+            mask, (self.__native_namespace__().Series, Sequence)
+        ):
+            mask = [mask]
```

Comment:
In principle, it should be a bool or a sequence/series of boolean values, but since I am checking for sequence type, better safe than sorry!